### ISSUE TITLE
Undefined array inside if condition or loop incorrectly marks other uses as read

### ISSUE THREAD

**Issue by sirbrillig** [sirbrillig/phpcs-variable-analysis#210]
In the following example, we get a false negative:

```php
function ifConditionWithUndefinedArrayAssignment($first) {
  if ($first) {
    $things[] = 'person'; // undefined array variable
  }
  return $things; // undefined variable, but not reported
}
```

This one too:

```php
function loopAndPushWithUndefinedArray($parts) {
  while ($part = array_shift($parts)) {
    $suggestions[] = $part; // undefined array variable
  }
  return $suggestions; // undefined variable, but not reported
}
```

This is because https://github.com/sirbrillig/phpcs-variable-analysis/pull/205 made it so that future uses of a variable after an array push are marked as defined. They need to only do that within the current `if` or `else` or loop scope.

**Comment by sirbrillig** [comment]
Nevermind, I was mistaken. The behavior is correct, given that the variables in question might be defined (and we can't know if they are without full static analysis). It matches the same behavior for non-array variables.
